test.js

```diff
@@ -187,3 +187,15 @@ test('new stack format on Node.js 15 and later', t => {
     at Object.<anonymous> (/home/fengkx/projects/test/stack.js:14:1)`;
 	t.is(cleanStack(stack), expected);
 });
+
+test('handle undefined', t => {
+	const stack = undefined;
+	const expected = undefined;
+	t.is(cleanStack(stack, {pretty: true}), expected);
+});
+
+test('handle not string', t => {
+	const stack = {};
+	const expected = undefined;
+	t.is(cleanStack(stack, {pretty: true}), expected);
+});
```

Comment:
I think this shouldn't be supported. Is there any real situation in which `.stack` is neither a string nor undefined?

Reply:
This pr is aimed to support the Error interface in typescript, which has `stack?: string`:
https://github.com/microsoft/TypeScript/blob/6df16b35f2bc5a53599f6f5f56dfd3f79545bd1c/lib/lib.es5.d.ts#L975

So that you can safely pass `error.stack` to this function even in `strict` mode.

> Is there any real situation in which .stack is neither a string nor undefined?

I don't know either.

Reply:
I think typescript has a reason -- [the `stack` property is non-standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack).

Reply:
I know, I think it's great to support `stack?: string`, which means supporting `string | undefined`.

What I'm saying is that I don't think you had to add a test for what happens with `stack = {}` (neither string nor undefined). I think you could assume it was either a string or absent. But now that you already did it, let's wait for what Sindre thinks.